components/access-management/kfam/bindings.go

```diff
@@ -90,9 +94,7 @@ func getAuthorizationPolicy(binding *Binding, userIdHeader string, userIdPrefix
 				},
 				From: []*istioSecurity.Rule_From{{
 					Source: &istioSecurity.Source{
-						Principals: []string{
-							"cluster.local/ns/istio-system/sa/istio-ingressgateway-service-account",
-						},
+						Principals: []string{istioIGWPrincipal},
```

Comment:
You also need to add the `kfpUIPrincipal` here to allow the KFP UI to proxy requests for non-owners.

So if we define `kfpUIPrincipal` like in the profile controller, you can simply extend this list:

```go
				From: []*istioSecurity.Rule_From{{
					Source: &istioSecurity.Source{
						Principals: []string{
							// end-users access via the ingress gateway
							istioIGWPrincipal,
							// kfp proxies user requests to artifact pods in profile namespaces
							kfpUIPrincipal,
						},
					},
				}},
```